gadget/update.go

```diff
@@ -258,9 +262,37 @@ func EnsureLayoutCompatibility(gadgetLayout *LaidOutVolume, diskLayout *OnDiskVo
 		// size and offset, so the last thing to check is that the filesystem
 		// matches (or that we don't care about the filesystem).
 
-		// TODO: here we need to handle in the strict case partitions which are
-		// to be created at install and are encrypted like ubuntu-data as they
-		// will not match filesystems exactly and need some massaging
+		// first handle the strict case where this partition was created at
+		// install in case it is an encrypted one
+		if opts.AssumeCreatablePartitionsCreated && IsCreatableAtInstall(gv) {
+			// only partitions that are creatable at install can be encrypted,
+			// check if this partition was encrypted
+			if encType, ok := opts.EncryptedPartitions[gs.Name]; ok {
+				switch encType {
+				case EncryptionICE:
+					return false, "inline cryptography engine encrypted partitions currently unsupported"
```

Comment:
I'd suggest starting the sentence with `Inline Crypto Engine` or even just ICE, because I googling for "inline cryptography engine" does not lead to sure results.